aspnet/hosting/dataprotection.rst

```diff
@@ -0,0 +1,21 @@
+.. _dataprotection:
+
+Data Protection
+===============
+
+By `Sourabh Shirhatti`
+
+The ASP.NET 5 data protection stack provides a simple and easy to use cryptographic API a developer can use to protect data, including key management and rotation. This document provides an overview of how to configure Data Protection on your server to enable developers to use data protection.
+
+Create Data Protection Registry Hive
+------------------------------------
+
+By default, keys are not persisted outside of the current process. When the process shuts down, all generated keys will be lost. To persist keys for an application hosted in IIS, you must create registry hives for each application pool to store the keys. You should use the `Provisioning PowerShell script <https://github.com/aspnet/DataProtection/blob/dev/Provision-AutoGenKeys.ps1>`_ for each application pool you will be hosting ASP.NET 5 applications under. This script will create a special registry key in the HKLM registry that is ACLed only to the worker process account. Keys are encrypted at rest using DPAPI.
+
+
+Machine Wide Policy
+-------------------
+
+The data protection system has limited support for setting default machine-wide policy for all applications that consume the data protection APIs. For more information on how to configure the machine wide policy have a look at :ref:`this article <data-protection-configuration-machinewidepolicy>`.
+
+
```

Comment:
I'd consider adding a section on how developers can choose where keys are stored, and how hosters should consider offering a directory outside of the application root (and properly ACLed), and the ability to upload X509 Certificates to protect file system key stores. Such certificates must be placed in the trusted certificate store see http://blogs.technet.com/b/sbs/archive/2007/04/10/installing-a-self-signed-certificate-as-a-trusted-root-ca-in-windows-vista.aspx and the private key ACLed to the app pool ID - https://technet.microsoft.com/en-us/library/ee662329.aspx?f=255&MSPPError=-2147217396